api/recommend.js

```diff
@@ -0,0 +1,68 @@
+const express = require('express');
+const asyncHandler = require('express-async-handler');
+const recombee = require('recombee-api-client');
+const crypto = require('crypto');
+
+const rqs = recombee.requests;
+const recombeeClient = new recombee.ApiClient(process.env.RECOMBEE_DATABASE_ID,
+  process.env.RECOMBEE_PRIVATE_TOKEN);
+
+
+const router = express.Router();
+
+router.post('/personal', asyncHandler(async (request, response, next) => {
+  const {
+    count = 10,
+  } = request.body;
+
+  if (!request.openid) {
+    response.sendStatus(403);
+    return;
+  }
+
+  const userHash = crypto.createHash('sha256').update(request.openid.user.sub).digest('hex');
+  await recombeeClient.send(
+    new rqs.RecommendItemsToUser(userHash, count, { scenario: 'personal_view' }),
+  )
+    .then((recommendation) => {
+      response.send({ recipes: recommendation.recomms.map((e) => e.id) || [] });
+    })
+    .catch(() => {
+      recombeeClient.send(
+        new recombee.RecommendItemsToUser(userHash, count, { scenario: 'homepage_view' }),
+      );
+    })
+    .then((recommendation) => {
+      response.send({ recipes: recommendation.recomms.map((e) => e.id) || [] });
+    })
+    .catch((err) => {
+      if (err) next(err);
+    });
+}));
```

Comment:
I'm not sure what this is for. Can you explain?

Reply:
The behaviour I tried creating was:
Try and get personal view
if successful: return recipes
else: try getting homepage view
if successful: return recipes
else: handle error

in general when using the asyncHandler we need to pass errs to the handler by calling next(err). otherwise errors cause the server to exit which is 💩

Reply:
amazing. I get it. Looks good to me